pkg/agent/multicluster/mc_route_controller.go

```diff
@@ -221,33 +232,215 @@ func (c *MCRouteController) Run(stopCh <-chan struct{}) {
 
 // worker is a long-running function that will continually call the processNextWorkItem
 // function in order to read and process a message on the workqueue.
-func (c *MCRouteController) worker() {
+func (c *MCDefaultRouteController) worker() {
 	for c.processNextWorkItem() {
 	}
 }
 
-func (c *MCRouteController) processNextWorkItem() bool {
+func (c *MCDefaultRouteController) processNextWorkItem() bool {
 	obj, quit := c.queue.Get()
 	if quit {
 		return false
 	}
 	defer c.queue.Done(obj)
 
-	if k, ok := obj.(string); !ok {
+	key, ok := obj.(string)
+	if !ok {
 		c.queue.Forget(obj)
 		klog.InfoS("Expected string in work queue but got", "object", obj)
 		return true
-	} else if err := c.syncMCFlows(); err == nil {
-		c.queue.Forget(k)
+	}
+	if c.wireGuardConfig != nil {
+		wireGuardInitialized := c.wireGuardInitialized
+		if err := c.reconcileWireGuard(key); err != nil {
+			klog.ErrorS(err, "Failed to reconcile Multicluster WireGuard")
+			c.queue.AddRateLimited(key)
+			return true
+		}
+		if !wireGuardInitialized && c.wireGuardInitialized {
+			klog.InfoS("The Gateway has been updated")
+			return true
+		}
+	}
+
+	if err := c.syncMCFlows(); err == nil {
+		c.queue.Forget(key)
 	} else {
 		// Put the item back on the workqueue to handle any transient errors.
-		c.queue.AddRateLimited(k)
-		klog.ErrorS(err, "Error syncing key, requeuing", "key", k)
+		c.queue.AddRateLimited(key)
+		klog.ErrorS(err, "Error syncing key, requeuing", "key", key)
 	}
 	return true
 }
 
-func (c *MCRouteController) syncMCFlows() error {
+// reconcileWireGuard reconcile WireGuard configurations in following way:
+//  1. If the key is a Gateway dummy key, it means that we should check current Node and initialize/remove corresponding
+//     WireGuard interface on the Gateway Node.
+//  2. If the key is not a Gateway dummy key, it means that a new ClusterInfoImport event comes. The controller should
+//     update WireGuard peers on the Gateway Node.
+func (c *MCDefaultRouteController) reconcileWireGuard(key string) error {
+	c.wireGuardMutex.Lock()
+	defer c.wireGuardMutex.Unlock()
+
+	gateway, err := c.getActiveGateway()
+	if err != nil {
+		klog.ErrorS(err, "Failed to get active Gateway")
+		return err
+	}
+
+	isCIImport := strings.HasSuffix(key, "clusterinfo")
+	if isCIImport {
+		if gateway == nil || gateway.Name != c.nodeConfig.Name {
```

Comment:
Again, I suggest you to add a boolean like `isMyselfGateway := gateway != nil && gateway.Name == c.nodeConfig.Name`.

Reply:
Done.